src/components/App.tsx

```diff
@@ -638,13 +638,16 @@ export class App extends React.Component<any, AppState> {
       );
       event.preventDefault();
     } else if (
-      shapesShortcutKeys.includes(event.key.toLowerCase()) &&
       !event.ctrlKey &&
       !event.altKey &&
       !event.metaKey &&
       this.state.draggingElement === null
     ) {
-      this.selectShapeTool(shape);
+      if (shapesShortcutKeys.includes(event.key.toLowerCase())) {
+        this.selectShapeTool(shape);
+      } else if (event.code === "KeyQ") {
+        this.toggleLock();
+      }
```

Comment:
Were you solving something by using `event.code`? Otherwise, I'd drop it because it may mess up keyboards with different layouts (`event.code` doesn't take layout into account) --- meaning some layouts may have a different printable key under `KeyQ` physical location), and thus could conflict with our other single-letter shortcuts.

Reply:
Sorry, I misinterpreted the code. You shouldn't "drop it". Just replace it with `event.key === "q"`.

Reply:
case in point: on AZERTY (french) keyboards, this PR doesn't work because pressing it triggers `(A)rrow` tool :).

Reply:
Fixed

Reply:
Thanks for the details!